### ISSUE TITLE
Support for GCC 9 ?!

### ISSUE THREAD

**Issue by soeth16** [bblanchon/ArduinoJson#1184]
Hi at all, 
In order to compile the framework with GCC 9.2 I made following changes:

src/ArduinoJson/Object/MemberProxy.hpp
```C++
...
template <typename TObject, typename TStringRef>
class MemberProxy : public VariantOperators<MemberProxy<TObject, TStringRef> >,
                    public Visitable {
  typedef MemberProxy<TObject, TStringRef> this_type;

 public:
  MemberProxy(const MemberProxy&) = default; // GCC 9
...
```

src/ArduinoJson/Array/ElementProxy.hpp
```C++
...
template <typename TArray>
class ElementProxy : public VariantOperators<ElementProxy<TArray> >,
                     public Visitable {
  typedef ElementProxy<TArray> this_type;

 public:
  ElementProxy(const ElementProxy&) = default; // GCC 9
...
```

extras/tests/JsonDocument/BasicJsonDocument.cpp:
```C++
...
class SpyingAllocator {
 public:
  SpyingAllocator(const SpyingAllocator&) = default; // GCC 9
...
```

**Comment by bblanchon** [comment]
Hi @soeth16,

Thank you for reporting this.
I fixed this warning in the main branch.

Best regards,
Benoit

**Comment by bblanchon** [comment]
This fix was included in [ArduinoJson 6.15.0](https://arduinojson.org/news/2020/03/22/version-6-15-0/).
